<a href="https://colab.research.google.com/github/guiccosta/machine-learning/blob/master/lista1_eco.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [ ]:
import numpy as np

In [ ]:
values = np.array(list(range(1,7)))

prob_Xx =  1/6

def E(values):

  """
  values: array de valores assumidos pela variável\n
  return: média analítica
  """
  
  global prob_Xx
  return np.sum(values)*prob_Xx

def Var(values):
  
  """
  values: array de valores assumidos pela variável\n
  return: variância analítica
  """
  
  global prob_Xx
  inner = np.power(values - E(values),2)
  return E(inner)

print('A Esperança de X é: {} e a Variância de X é: {}'.format(E(values),Var(values)))

A Esperança de X é: 3.5 e a Variância de X é: 2.9166666666666665


In [ ]:
def simulation(n):

  """ n: numero de lançamentos de um dado não viciado \n
      return: media amostral e variância amostral 
  """
  
  s = np.random.choice(list(range(1,7)),size = n)
  return (s.mean(),s.var())

In [ ]:
print('Para (n = 10): {} \nPara (n = 100): {}\nPara (n = 100000): {}'.format(simulation(10), simulation(100), simulation(100000)))

Para (n = 10): (3.3, 2.6100000000000003) 
Para (n = 100): (3.44, 2.9463999999999997)
Para (n = 100000): (3.50178, 2.9157768316000006)



# **Respostas**

###**Questão 1)**###

**a)** A média analítica e a variância analítica de um dado de 6 faces não viciado é, respecitivamente, 3.5 e 2.9167, aproximadamente.

**b)** Para n = 10, a média e variância amostrais são, respectivamente, 3.3 e 2.6100000000000003, aproximadamente. Como podemos ver, tanto a média, quanto a variância amostral, diferem dos valores encontrados no item a.

**c)** Para n = 100, a média e variância amostrais são, respectivamente, 3.44 e 2.9463999999999997, aproximadamente. Já, para n = 100000, a média e variância amostrais são, respectivamente, 3.50178 e 2.9157768316000006, aproximadamente. Podemos ver que, à medida que n cresce, os valores da média amostral e da variância amostral convergem para os valores analíticos. Isso se dá devido a "*Lei dos Grandes Números*", que garante que quanto mais repetições de uma variável aleatória tivermos, mais a média amostral se aproxima do valor real da média analítica. 



---



---



In [ ]:
import scipy.integrate as integrate

In [ ]:
def N(x,mu = 0,sigma2 = 1):
  """
  x: array de valores reais \n
  mu: média \n
  sigma2: desvio padrão \n\n

  Calcula o valor da função densidade no ponto x
  """
  exp = np.exp(-0.5 * np.power(x - mu,2)/sigma2)
  return 1/(np.sqrt(2*np.pi))*exp

In [ ]:
print('A probabilidade que a variável aleatória X esteja entre -1.96 e 1.96 é {}'.format(integrate.quad(N, a = -1.96, b = 1.96)[0]))

A probabilidade que a variável aleatória X esteja entre -1.96 e 1.96 é 0.9500042097035591


In [ ]:

print('A probabilidade que a variável aleatória X seja menor que -1.96 e maior que 1.96 é aproximadamente {}'.format(integrate.quad(N, a = -np.inf, b = -1.96)[0] + integrate.quad(N, a = 1.96, b = np.inf)[0]))

A probabilidade que a variável aleatória X seja menor que -1.96 e maior que 1.96 é aproximadamente 0.049995790296433154


In [ ]:
def N1(x,mu = 0,sigma2 = 1):
  """
  x: array de valores reais \n
  mu: média \n
  sigma2: desvio padrão \n

  Função para auxiliar no cálculo da esperança de uma variável aleatória com distribuição normal.\n
  Retorna o valor da função densidade aplicada em x, tudo isso vezes x
  """
  return N(x)*x

In [ ]:
esperanca = integrate.quad(N1, a = -np.inf, b = np.inf)[0]

esperanca

0.0

In [ ]:
def N2(x,mu = 0,sigma2 = 1):
  """
  x: array de valores reais \n
  mu: média \n
  sigma2: desvio padrão \n\n

  Função para auxiliar no cálculo da variancia de uma variável aleatória com distribuição normal.\n
  Retorna o valor da função densidade aplicada em x, tudo isso vezes x ao quadrado.
  """

  exp = np.exp(-0.5 * np.power(x - mu,2)/sigma2)
  return 1/(np.sqrt(2*np.pi))*exp*np.power(x,2)

In [ ]:
variancia = integrate.quad(N2, a = -np.inf, b = np.inf)[0] + np.power(esperanca,2)

variancia

1.000000000000001

In [ ]:
print('A esperança e variância de X é, respectivamente, {} e {}, aproximadamente'.format(esperanca, variancia))

A esperança e variância de X é, respectivamente, 0.0 e 1.000000000000001, aproximadamente


# **Respostas**

### **Questão 2**

**a)** A probabilidade que a variável aleatória X esteja entre -1.96 e 1.96 é 0.9500042097035591.

**b)** A probabilidade que a variável aleatória X seja menor que -1.96 e maior que 1.96 é aproximadamente 0.049995790296433154

**c)** *Vide o código*